error_handler.py

```diff
@@ -40,21 +41,20 @@ async def dmauthor(*args, **kwargs):
                 logger.info(f"Reply to {ctx.message.id} and dm to {ctx.author.id} failed. Aborting.")
 
         async def reply(msg, file=None, embed=None):
-            if ctx.interaction:
-                if ctx.interaction.response.is_done():
-                    return await ctx.interaction.edit_original_response(content=msg,
-                                                                        attachments=[file] if file else None,
-                                                                        embed=embed)
+            try:
+                if ctx.interaction and ctx.interaction.response.is_done():
+                    return await ctx.interaction.edit_original_response(
+                        content=msg,
+                        attachments=[file] if file else None,
+                        embed=embed
+                    )
                 else:
                     return await ctx.reply(msg, file=file, embed=embed)
```

Comment:
Is avoiding the race condition better, or is the cost from the extra API call too much for the race condition?

(Please check the formatting before replying, I'm probably off by a space somewhere)

```suggestion
            try:
                return await ctx.interaction.edit_original_response(
                        content=msg,
                        attachments=[file] if file else None,
                        embed=embed
                    )
            except discord.NotFound:
                return await ctx.reply(msg, file=file, embed=embed)
```

Reply:
I don't think there is a race condition. Once an interaction is done, it can't be "undone", it's a permanent state.

Reply:
I was thinking "response was in-progress at the time of check, but then finished while making the call to `ctx.reply()`, leading to a double-reply". You'd know better than me, though. If that's not a risk, then I'm happy.